review verdict: changes requested (blocking), latest head `ea01c03870356f58f90c099a13546f9f2582d288`.

blocking finding:

- [p1] `server/sessions.ts:1237` — required CI is red because `npm run check` fails on an unused helper introduced/left behind by the `pendingSince` refactor. `isPendingSessionsFileStale()` is no longer referenced after restore switched to per-record `isPendingSessionStale()`, and eslint blocks the PR with `@typescript-eslint/no-unused-vars`. This is not product data loss, but it is a release blocker because #366 cannot land with failing required CI.

product re-check:

- the stale preserved-failure timestamp blocker is fixed on this head. Fresh live PTY records get `pendingSince: serializedAt`, the manifest `timestamp` is fresh, preserved failed records keep their original `pendingSince`/legacy manifest timestamp, and restore filters stale pending records per session before restoring. The regression test `serializeAll keeps fresh live sessions restorable when old preserved failures age out` covers the old-failure + fresh-live-session + threshold-crossing case.

bot feedback triage on latest head:

- Gemini inline `pruneScrollbackFiles` ENOENT/logging comment: fixed/noise (`server/sessions.ts:636-652`).
- Copilot inline `removeScrollbackDir` missing path in warning: fixed/noise (`server/sessions.ts:675-680`).
- CodeRabbit major finding about fresh sessions inheriting preserved failure timestamp: stale/fixed on latest code (`server/sessions.ts:765-785`, `1231-1257`, `1376-1407`).
- CodeRabbit tmp filename collision/naming/test nits: follow-up/nit only, not release-blocking.

verification checked:

- PR head changed from `ff94600f9bbe544fbd0f24c59c42155458a8fd94` to `ea01c03870356f58f90c099a13546f9f2582d288`.
- `gh pr checks 380`: `ci` fails, `e2e` skipped due failed dependency, `scan-commits` and CodeRabbit pass.
- CI failure log and local `rtk npm run check` both show `server/sessions.ts:1237:10 error 'isPendingSessionsFileStale' is defined but never used`.
- focused timestamp regression tests pass: `rtk npm test -- test/sessions.test.ts -t "serializeAll keeps fresh live sessions restorable|serializeAll preserves failed restore records|serializeAll prunes stale failed restore records|restoreFromDisk ignores stale files|restoreFromDisk removes malformed timestamp"`.

follow-up: created backend fix task `t_b0a435f0` and re-review task `t_c1007952`; linked the re-review before release gate `t_7aa1bfae`.

--- a/server/sessions.ts
+++ b/server/sessions.ts
@@ -1234,11 +1234,6 @@ function pendingSessionsAgeMs(pending: PendingSessionsFile): number | null {
   return Date.now() - timestampMs;
 }
 
-function isPendingSessionsFileStale(pending: PendingSessionsFile): boolean {
-  const ageMs = pendingSessionsAgeMs(pending);
-  return ageMs === null || ageMs > STALE_THRESHOLD_MS;
-}
-
 function pendingSessionAgeMs(
   pending: PendingSessionsFile,
   session: SerializedPtySession

